packages/atomic/src/components/search/atomic-generated-answer/atomic-source-citation.tsx

```diff
@@ -0,0 +1,73 @@
+import {
+  GeneratedAnswerCitation,
+  InteractiveCitation,
+  buildInteractiveCitation,
+} from '@coveo/headless';
+import {Component, Prop, Element, h} from '@stencil/core';
+import {Bindings} from '../../../components';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {buildCustomEvent} from '../../../utils/event-utils';
+import {LinkWithResultAnalytics} from '../../common/result-link/result-link';
+
+@Component({
+  tag: 'atomic-source-citation',
+  shadow: false,
+})
+export class AtomicSourceCitation implements InitializableComponent<Bindings> {
```

Comment:
I knew this was coming 😛 
I made a web component because function components are stateless so AFAIK they aren't suitable for managing a controller instance.
The `atomic-smart-snippet-source` (also internal) is the precedent I used to allow myself to make this a web component.
What do you think though?
Unless you know a way the function component can properly manage its own controller instance, then I think the only other way is the higher level `atomic-generated-answer` component will need to instantiate and hold references for all 1-5 interactive citation controllers. I prefer a web component with a single responsibility over that.

Reply:
Alright ;) 

What about the need to configure how the link behave ?

Reply:
Are you convinced or just don't want to argue? 😅
I'm not trying to be forceful btw I just genuinely don't see a better solution. Very open to one though.

As for the configuration, the citations are going to get an update in the near future on 2 fronts:
1. On hover they will display a popover card with some more expanded details about the snippet. Jury is still out on whether that will be configurable.
2. We need to find a solution to allow some kind of hreftemplate for things like salesforce links. The solution for that is TBD.
In the future there might be a public component related to citations but for now it remains private. Just trying to fix the analytics issue here.

Reply:
By making a web component do you feel like we're taking a step we can't walk back? Given it's @internal I thought we were in a safe space here.

Reply:
It's not marked @internal though, since it generated new symbols for angular. 
With that fixed, yes, it's safer.

I think the best would be if there was no such multiple layers of indirection/things were "flatter" at the top level, and more of the work was visible from the top level public component.


We have -> 
`atomic-generated-answer` -> `GeneratedContent` -> `SourceCitations` -> Loop over citations -> `atomic-source-citations`

A possible alternative could be something like (pseudo code...): 

```typescript
// atomic-generated-answer.tsx

renderContent() {
    const {answer, citations} = this.generatedAnswerState;
     // ....
     <GeneratedContentContainer {...this.generatedAnswerState} >
         {answer}
         <CitationsContainer label={this.bindings.i18n.t('more-info')}>
              {citations.map(citation=> {
                   const interactiveCitation = buildInteractiveCitation(this.bindings.engine, {options: {citation}})
                   return <LinkWithResultAnalytics href={citation.uri} etc.. />
              })
         </CitationsContainer>
     </GeneratedContentContainer>
 
}
```

Reply:
And with the fact that those functional component would not be re-used anywhere else (ie: they're pretty much just private function for `atomic-generated-answer`), they should also just be that, a private function on the class.

Reply:
Wow! The fact that i forgot to mark it internal is super embarrassing.
.
.
.
in your snippet wouldn't the controller be reinstantiated every time the citations container renders?

Reply:
Any time the citations changes, yes (which is the same as right now ? 🤔). 

Again, it's just pseudo code, and you'd certainly want to mimic what's already here (ie: render citation only when it finishes streaming).

The goal of the comment is not "copy paste exactly this", but the general idea.

The red flag with how it is right now is that we have to create a global component (but mark it internal) mostly because of how the code is internally structured (there's ~3 level of encapsulation that passes properties down), but for no "actual good product reason".

An internal web component can have it's place if it gets re-used in multiple place in the project, which is currently not the case here.

Reply:
All right I can find a way around the internal web component since it seems to be a deal breaker. No problem. 👍 
I refactored the encapsulation yesterday afternoon. Do you still think it's poorly structured?

Reply:
Updated it again just now.